tap_kit/client.py

```diff
@@ -31,18 +33,26 @@ def requests_method(method, request_config, body):
 
     @backoff.on_exception(backoff.expo,
                           RateLimitException,
+                          Exception,
                           max_tries=10,
                           factor=2)
     def make_request(self, request_config, body=None, method='GET'):
         LOGGER.info("Making {} request to {}".format(
             method, request_config['url']))
 
         with singer.metrics.Timer('request_duration', {}) as timer:
-            response = self.requests_method(method, request_config, body)
+            try:
+                response = self.requests_method(method, request_config, body)
+            except Exception:
+                raise Exception()
+
 
         if response.status_code in [429, 503]:
             raise RateLimitException()
 
+        if response.status_code in [400, 502, 504]:
```

Comment:
I would create 2 lists, one for `RATE_LIMIT_ERROR_CODES` and another for `OTHER_RETRYABLE_ERROR_CODES` (you can tweak the name if they fit into a neat category. 

Then have one `if` statement:
```if response.status_code in RATE_LIMIT_ERROR_CODES + OTHER_RETRYABLE_ERROR_CODES:
    raise RetryableException()  # rename the RateLimitException```

Reply:
Made the changes you suggested. Do you mind taking another look at it? Also what is the best way for me to test this so I am confident it won't break things before I merge the changes?